src/json-helper.c

```diff
@@ -72,18 +72,28 @@ static JsonNode* json_get_first_matching_element(JsonNode *json_node,
         return g_steal_pointer(&node);
 }
 
-gchar* json_get_string(JsonNode *json_node, const gchar *path)
+gchar* json_get_string(JsonNode *json_node, const gchar *path, GError **error)
 {
-        char *res_str = NULL;
-        JsonNode *result = json_path_query(path, json_node, NULL);
-        if (result) {
-                JsonArray *results = json_node_get_array(result);
-                if (json_array_get_length( results ) > 0) {
-                        res_str = g_strdup(json_array_get_string_element(results, 0));
-                }
+        g_autofree char *res_str = NULL;
```

Comment:
If we use glib cleanup here, we should also change to glib type here (`gchar`).

Reply:
Done.